tests/virtual_machines_test.go

```diff
@@ -489,6 +489,67 @@ var _ = Describe("Virtual Machines", func() {
 				Expect(err).ToNot(HaveOccurred())
 			})
 		})
+
+		Context("When a VM's NIC is removed and a new VM is created with the same MAC", func() {
+			It("should successfully release the MAC and the new VM should be created with no errors", func() {
+				err := setRange(rangeStart, rangeEnd)
+				Expect(err).ToNot(HaveOccurred())
+
+				vm := CreateVmObject(TestNamespace, false, []kubevirtv1.Interface{newInterface("br1", ""), newInterface("br2", "")},
+					[]kubevirtv1.Network{newNetwork("br1"), newNetwork("br2")})
+				Eventually(func() error {
+					return testClient.VirtClient.Create(context.TODO(), vm)
+
+				}, 50*time.Second, 5*time.Second).Should(Not(HaveOccurred()), "failed to apply the new vm object")
+				mac1, err := net.ParseMAC(vm.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress)
+				Expect(err).ToNot(HaveOccurred())
+				mac2, err := net.ParseMAC(vm.Spec.Template.Spec.Domain.Devices.Interfaces[1].MacAddress)
+				Expect(err).ToNot(HaveOccurred())
+
+				By("before the NIC is deleted")
+				newVm := CreateVmObject(TestNamespace, false, []kubevirtv1.Interface{newInterface("br1", mac1.String())},
+					[]kubevirtv1.Network{newNetwork("br1")})
+				Eventually(func() error {
+					err = testClient.VirtClient.Create(context.TODO(), newVm)
+					if err != nil && strings.Contains(err.Error(), "failed to allocate requested mac address") {
+						return err
+					}
+					return nil
+
+				}, 50*time.Second, 5*time.Second).Should(HaveOccurred(), "should have failed to create the vm because the mac is allready taken")
+
+				By("deleting the VM's NIC")
+				Eventually(func() error {
+					err := testClient.VirtClient.Get(context.TODO(),
+						client.ObjectKey{Namespace: vm.Namespace, Name: vm.Name}, vm)
+					if err != nil {
+						return err
+					}
+
+					vm.Spec.Template.Spec.Domain.Devices.Interfaces = []kubevirtv1.Interface{newInterface("br2", mac2.String())}
+					vm.Spec.Template.Spec.Networks = []kubevirtv1.Network{newNetwork("br2")}
+					err = testClient.VirtClient.Update(context.TODO(), vm)
+					if err != nil {
+						return err
+					}
+
+					if len(vm.Spec.Template.Spec.Domain.Devices.Interfaces) != 1 {
+						return fmt.Errorf("failed to delete the VM's interface")
+					}
+
+					return nil
+				}, 50*time.Second, 5*time.Second).Should(Not(HaveOccurred()), "failed to update client")
+
+				By("after the VM's NIC has been deleted")
+				Expect(vm.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress).To(Equal(mac2.String()),
+					"kubemacpool should assign the same mac as before the update")
```

Comment:
This is not a good test. It assumes that we always use the first available mac from the pool, and that is undocumented and can change any time. I mean, it does the job, but is not really what we want to test. Could we make it, so our pool has only 2 available addresses, if one from removed NIC has been recycled, we should be able to create a VM here and get **a** MAC. Sorry if I'm nitpicking too much.

Reply:
HI @phoracek .It's all good :)
I didn't understand though whats the point of limiting the range if we are not checking kubemacpool assigns the same mac that was released (the only one that is available).
If we are checking that the new Vm received just "a MAC" than maybe we can leave the range as it is? 
Thanks:)

Reply:
This test schould check "do we release MAC after this and that happens with a VM". What you wrote checks that, but also "we always assign the first MAC address from the pool". The latter check is not documented and irrelevant.

I was thinking how could we check 1. without 2. What came to my mind is, let's limit the pool to two MAC addresses during the deployment. Then if the MAC is not released, we fail to allocate, if we release, we will have a MAC address to assign. Same results, less assumptions.

Does that make sense? Trying to do BDD testing, hehe.

Anyway, right now we don't control the range while testing, do we? It is only configured once during the deployment, isn't it?

Reply:
we can do it

```golang
err := setRange(rangeStart, rangeEnd)
```

 we can put a range just before any test

Reply:
> Does that make sense? Trying to do BDD testing, hehe.
OK. I see your point here.
NP I'll change that.
> Anyway, right now we don't control the range while testing, do we? It is only configured once during the deployment, isn't it?
I think we can. The "setRange" function sets the range for the macpool by configuring the range and deleting the kubemacpool pod and waiting for it to run again.

Reply:
I think it would be the right way to do it.